libcontainer/container_linux.go

```diff
@@ -201,24 +201,23 @@ func (c *Container) Set(config configs.Config) error {
 
 // Start starts a process inside the container. Returns error if process fails
 // to start. You can track process lifecycle with passed Process structure.
-func (c *Container) Start(process *Process) error {
+func (c *Container) Start(process *Process) (retErr error) {
 	c.m.Lock()
 	defer c.m.Unlock()
 	if c.config.Cgroups.Resources.SkipDevices {
 		return errors.New("can't start container with SkipDevices set")
 	}
 	if process.Init {
+		defer func() {
+			if retErr != nil {
+				c.deleteExecFifo()
+			}
+		}()
 		if err := c.createExecFifo(); err != nil {
```

Comment:
IMHO, it seems cleaner if `createExecFifo()` deletes it in case of error. That way, callers only need to delete it if that function succeeded and some other error was found and the function decides to rollback the changes made.

It seems error-prone and hard to catch on review if callers of the function need to be aware that on failure you need to delete it and handle it yourself (as not all the functions are like that).

Reply:
Make sense, thanks.